Add a page on TSC responsibilities (#31)
hyperledger/tsc: tsc-responsibilities.md

---
layout: default
title: TSC responsibilities
nav_order: 21
---
[//]: # (SPDX-License-Identifier: CC-BY-4.0)

# The Hyperledger Technical Steering Committee (TSC) responsibilities

> The [Hyperledger Charter]() states that the TSC is responsible for:

Reviewer: ```suggestion
The [Hyperledger Charter](https://www.hyperledger.org/about/charter) states that the TSC is responsible for:
```

1. coordinating the technical direction of the Hyperledger Foundation;

2. approving project proposals (including, but not limited to, incubation, deprecation and changes to a project’s charter or scope) in accordance with a project lifecycle document to be developed, approved and maintained by the TSC;

3. creating sub-committees or working groups to focus on cross-project technical issues or opportunities;

4. communicating with external and industry organizations concerning related technical matters;

5. appointing representatives to work with other open source or standards communities;

6. establishing community norms, workflows, or policies for releases;

7. discussing, seeking consensus, and where necessary, voting on technical matters relating to the code base that affect multiple projects; and

8. establishing election processes for Maintainers or other leadership roles in the technical community that are not within the scope of any single project.

## TSC Members

TSC members are expected to:

> * Regularly participate in the [TSC Meetings](https://wiki.hyperledger.org/display/TSC/TSC+Meeting+Minutes)

Reviewer 2: How about adding a line or two, that captures how TSC members can help to grow the community. I have tried to capture my thoughts in this following paragraph

TSC members are stewards for Hyperledger Foundation, they represent a good portion of our community. They are responsible for growing a healthy community/ecosystem. This would include resolving conflicts where necessary, promoting the blockchain/multi-party system technologies, identifying gaps in the current implementation, being ears to what's happening in the space from similar parties outside Hyperledger.

Reviewer 3: @arsulegai  this  really great idea. As a TSC member we are brand ambassador for the hyperledger foundation to the community out there.

Author: I think it's a good idea to add something about acting as an ambassador or steward for the organization but given that this was never spelled out I wonder whether every TSC member expects this to be part of their role. It'd be good to hear from everybody on this.

Author: Done.

Reviewer 2: Thanks

* Review the [TSC Project Updates](https://wiki.hyperledger.org/display/TSC/TSC+Project+Updates) and comment as needed in a timely manner
* Bring up and help resolve any issues related to the needs of the Hyperledger technical community

> * Participate in the [Taskforces](https://wiki.hyperledger.org/display/TF/Task+Forces+Home) set up by the TSC to address specific issues

Reviewer: ```suggestion
* Participate in, and optionally chair, the [Task Forces](https://wiki.hyperledger.org/display/TF/Task+Forces+Home) set up by the TSC to address specific issues
```

> * Attend one project meeting a month for a project with which they are unaffiliated and have not been a contributor (and, ideally, one that they have not attended before). See [TSC Issue #21](https://github.com/hyperledger/tsc/issues/21)).

Reviewer 2: Do we really need to call this out?

Reviewer: Yes, we agreed as a TSC that this was a requirement for TSC members. Why would we not call this out?

Author: I also don't see why not.

Reviewer 2: I am inclined towards adding the motive behind why we arrived at this resolution. Today the motive is served through attending the meeting once a month, it could be different in future.

Author: > I am inclined towards adding the motive behind why we arrived at this resolution. Today the motive is served through attending the meeting once a month, it could be different in future.

@arsulegai I think if you feel this is needed it would be better to add that to the issue itself. I don't think this document should explain the rationale for everything. I'd rather keep it a tight list. Anyone who wants to know more can follow the link to the issue.

## TSC Chair

The TSC Chair has the following additional responsibilities:

* Running the TSC meetings, such as the TSC calls per the agreed upon schedule. This includes: setting up and publishing an agenda, running the meeting, and ensuring any outcome is duly recorded
* Representing the TSC, and more broadly the Hyperledger technical community, on the [Governing Board](https://www.hyperledger.org/about/leadership#governing), and give updates to the Governing Board on TSC activities

> * Giving a TSC update at the [Hyperledger Global Forum](https://events.linuxfoundation.org/hyperledger-global-forum/)

Reviewer: ```suggestion
* Giving a TSC update at the [Hyperledger Global Forum](https://events.linuxfoundation.org/hyperledger-global-forum/)
* Giving a TSC update at the Hyperledger Member Summit
```

> * Representing the Hyperledger technical community externally, including presenting at conferences and responding to press inquiries, in coordination with the Hyperledger staff

Reviewer: ```suggestion
* Representing the Hyperledger technical community externally, including presenting at conferences and responding to press inquiries, in coordination with the Hyperledger staff

## TSC Vice Chair
The TSC Vice Chair has the following additional responsibilities:

* Running the TSC meetings when the chair is unable
* Reaching out to projects when they miss submitting their TSC Quarterly Project Update to remind the projects that they have a report due
```

---
layout: default
title: TSC responsibilities
nav_order: 21
---
[//]: # (SPDX-License-Identifier: CC-BY-4.0)

# The Hyperledger Technical Steering Committee (TSC) responsibilities

The [Hyperledger Charter](https://www.hyperledger.org/about/charter) states that the TSC is responsible for:

1. coordinating the technical direction of the Hyperledger Foundation;

> 2. approving project proposals (including, but not limited to, incubation, deprecation and changes to a project’s charter or scope) in accordance with a project lifecycle document to be developed, approved and maintained by the TSC;

Reviewer 4: does it also include labs?

Author: No. Labs have been designed to provide a space that's explicitly not under direct governance of the TSC and do not imply endorsement from it.

> 3. creating sub-committees or working groups to focus on cross-project technical issues or opportunities;

Reviewer 4: I would change the wording here instead to change creating to supporting or emerging groups to focus on dedicated areas whenever needed. Cause otherwise, it makes an impression that TSC members ought to create WIG.

Author: As the text above that list indicates, this comes from the charter which is under the control of the Governing Board. We cannot change it ourselves. We can ask the board to update it if you think that's worth the effort.

> 4. communicating with external and industry organizations concerning related technical matters;

Reviewer 4: I do not have any particular comments about that one, but I'm just curious: is there a list of technical topics and corresponding industry organisations?

Author: I don't think so.

5. appointing representatives to work with other open source or standards communities;

6. establishing community norms, workflows, or policies for releases;

7. discussing, seeking consensus, and where necessary, voting on technical matters relating to the code base that affect multiple projects; and

8. establishing election processes for Maintainers or other leadership roles in the technical community that are not within the scope of any single project.

> 

Reviewer 4: I know this is kind of implicitly expected but probably worth adding here:

9. Subscribe to the tsc@ mailing list to stay aware of the TSC related updates and issues.

Author: I can add that to the second list.

## TSC Members

TSC members are expected to:

* Regularly participate in the [TSC Meetings](https://wiki.hyperledger.org/display/TSC/TSC+Meeting+Minutes)

> * Review the [TSC Project Updates](https://wiki.hyperledger.org/display/TSC/TSC+Project+Updates) and comment as needed in a timely manner

Reviewer 4: IMO, this should be moved into responsibilities along the lines, and I believe we should add something that reflects checking project health status and making sure that the report is accurate and complete.

Author: The first list comes from the charter while this one is from the TSC so I don't think moving it is an option, it would confuse things.

I'm happy to extend what the review of the reports entails but I don't think we can require TSC members "[make] sure that the report is accurate and complete". How would one do that? Of course a quick look at LF Insights might reveal some discrepancy with what the report claims but I don't know how one would ensure accuracy and completeness.

* Bring up and help resolve any issues related to the needs of the Hyperledger technical community
* Participate in, and optionally chair, the [Task Forces](https://wiki.hyperledger.org/display/TF/Task+Forces+Home) set up by the TSC to address specific issues

> * Attend one project meeting a month for a project with which they are unaffiliated and have not been a contributor (and, ideally, one that they have not attended before). See [TSC Issue #21](https://github.com/hyperledger/tsc/issues/21)).

Reviewer: ```suggestion
* Attend one project meeting a month for a project with which they are unaffiliated and have not been a contributor (and, ideally, one that they have not attended before). See [TSC Issue #21](https://github.com/hyperledger/tsc/issues/21).
```

Author: Indeed, I noticed that during the call too! :-)

> * Act as stewards for Hyperledger promoting and helping grow the organization and its activities by engaging of their own accord in activities such as posting on social media, responding to questions raised in forums, helping new community members find their way around, and giving talks at conference on Hyperledger related topics.

Reviewer 5: ```suggestion
* Act as stewards for Hyperledger promoting and helping grow the organization and its activities by engaging of their own accord in activities such as posting on social media, responding to questions raised in forums, helping new community members find their way around, and giving talks at conferences on Hyperledger related topics.
```
